### PR TITLE
Nymph2 2/python packaging framework

### PR DESCRIPTION
**Pull request by zieglerad**

### CODE REVIEW

File: Cpp/Library/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# CMakeLists for Nymph/Cpp/Library
+# CMakeLists for Nymph/Src/Cpp/Library
```
Comment (MCFlowMace) [inline Cpp/Library/CMakeLists.txt:1]: That change is not correct

Comment (zieglerad) [reply Cpp/Library/CMakeLists.txt]: Fixed this comment

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,20 @@
+[metadata]
+name = nymph
+version = 2.0.0
```
Comment (MCFlowMace) [inline setup.cfg:3]: I suggest to use this https://github.com/pypa/setuptools_scm instead. That way the version is always in sync with the git version tag.

File: CMakeLists.txt [hunk 3]
```diff
@@ -115,7 +123,7 @@ pbuilder_add_submodule( Scarab Scarab )
 pbuilder_use_sm_library( Scarab Scarab )
 
 # this is needed for anything that wants to read or write JSON files
-get_directory_property( RAPIDJSON_FILE_BUFFER_SIZE DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/Scarab DEFINITION RAPIDJSON_FILE_BUFFER_SIZE )
+get_directory_property( RAPIDJSON_FILE_BUFFER_SIZE DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} Scarab DEFINITION RAPIDJSON_FILE_BUFFER_SIZE )
```
Comment (MCFlowMace) [inline CMakeLists.txt:126]: Again do not see why that change was necessary. But again does not seem to matter.

**Review by MCFlowMace — APPROVED** [review]
A couple of comments/questions as seen above. But it seems to work fine. We can also look into using `setuptools_scm` but that is not necessary to merge.

File: setup.cfg (hunk 2 above)
Comment (nsoblath) [reply setup.cfg:3]: Fine with me to merge without this.

File: NymphConfig.cmake.in [hunk 4]
```diff
@@ -4,7 +4,7 @@ get_filename_component( Nymph_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH )
 
 # Find the dependencies
 include( CMakeFindDependencyMacro )
-find_dependency( Scarab REQUIRED PATHS ${Nymph_CMAKE_DIR}/Scarab @Scarab_BINARY_LOCATION@ )
+find_dependency( Scarab REQUIRED PATHS ${Nymph_CMAKE_DIR} /Scarab @Scarab_BINARY_LOCATION@ )
```
[the remark this answers is not in the document]
Comment (nsoblath) [reply NymphConfig.cmake.in:7]: This needs to be resolved.  I think it'll cause a problem with downstream packages (namely Katydid)

File: CMakeLists.txt (hunk 3 above)
Comment (nsoblath) [reply CMakeLists.txt:126]: This needs to be resolved too

**Review by nsoblath — CHANGES_REQUESTED** [review]
Just a couple minor things to resolve, which @MCFlowMace had pointed out.

**Review by nsoblath — APPROVED** [review]
Looks good

**Pull request merged**